src/Periphery/Permit2Proxy.sol

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import { ISignatureTransfer } from "permit2/interfaces/ISignatureTransfer.sol";
+import { TransferrableOwnership } from "lifi/Helpers/TransferrableOwnership.sol";
+import { LibAsset, IERC20 } from "lifi/Libraries/LibAsset.sol";
+import { PermitHash } from "permit2/libraries/PermitHash.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20Permit } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+
+/// @title Permit2Proxy
+/// @author LI.FI (https://li.fi)
+/// @notice Proxy contract allowing gasless (Permit2-enabled) calls to our
+///         diamond contract
+/// @custom:version 1.0.0
+contract Permit2Proxy {
+    /// Storage ///
+
+    address public immutable LIFI_DIAMOND;
```

Comment:
I am not sure if ti makes sense to have the diamond address permanently tied to the bytecode. That would require us to redeploy Permit2 with every new diamond deployment and I dont see any reason why this would be necessary. For this we have the diamondWhitelist.
In a future we we may have multiple diamonds, we would not be able to use this contract for all of them but would require multiple `Permit2` contracts. 
Is this required?

Reply:
I discussed this with @maxklenk and we decided multiple Permit2Proxies was preferable to maintaining another whitelist. We can discuss again in the sync.

Reply:
The users approval is given to a fixed Permit2 contract, which proxy the user calls to actually execute his transaction does not really matter for him. In order to keep the complexity low and have contracts immutable whenever possible I would prefer to keep the one-proxy-per-diamond setup.